### PR TITLE
feat: auto-join threads

### PR DESCRIPTION
**Pull request by DevChromium**

### CODE REVIEW

File: src/events/threadCreate.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            },
+        ])
+
+        thread.join();
+        
+        logChannel.send({embeds: [embed.toJSON()]})
+        consola.success(`Joined thread ${thread.name}`);
+    } catch(err: any){
+        consola.error(`Failed to join thread ${thread.name}: ${err}`)
+        logError(err);
```
Comment (DevScyu) [inline src/events/threadCreate.ts:36]: logError will also do a console output, is the line above needed?

File: src/events/threadCreate.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import consola from "consola";
+import { TextChannel, ThreadChannel } from "discord.js";
+import { logError, styledEmbed } from "../utils/functions";
+
+export const action = async (thread: ThreadChannel): Promise<void> => {
+    try {
+        const logChannel = thread.guild.channels.cache.find(c => c.name === "server-logs") as TextChannel
```
Comment (DevScyu) [inline src/events/threadCreate.ts:7]: might want to add a if(logChannel) before trying to send the message, as the channel could not exist

File: src/events/threadDelete.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            {
+                name: 'Closed at',
+                value: `<t:${Math.floor(Date.now() / 1000)}:R>`,
+                inline: true
+            },
+        ])
+        logChannel.send({embeds: [embed.toJSON()]})
+    } catch (err: any) {
+        consola.error(err)
+        logError(err)
```
Comment (DevScyu) [inline src/events/threadDelete.ts:32]: same as above

File: src/events/threadDelete.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import consola from "consola";
+import { TextChannel, ThreadChannel } from "discord.js";
+import { logError, styledEmbed } from "../utils/functions";
+
+export const action = async (thread: ThreadChannel) => {
+    try {
+        const logChannel = thread.guild.channels.cache.find(c => c.name === "server-logs") as TextChannel
```
Comment (DevScyu) [inline src/events/threadDelete.ts:7]: same as above, check if the channel actually exists

File: src/events/threadCreate.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                name: 'Owner',
+                value: `<@${thread.ownerId}>`,
+                inline: true
+            },
+            {
+                name: 'Created at',
+                value: `<t:${Math.floor(thread.createdTimestamp / 1000)}:R>`,
+                inline: true
+            },
+        ])
```
Comment (DevScyu) [inline src/events/threadCreate.ts:28]:
```suggestion
        const embed = styledEmbed()
            .setColor(0x72ed9e)
            .setTitle("Thread Notification")
            .setDescription("An thread has been created.")
            .addFields([
            {
                name: 'Name',
                value: thread.name,
                inline: true
            },
            {
                name: 'Owner',
                value: `<@${thread.ownerId}>`,
                inline: true
            },
            {
                name: 'Created at',
                value: `<t:${Math.floor(thread.createdTimestamp / 1000)}:R>`,
                inline: true
            },
        ])
```

File: src/events/threadDelete.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        const embed = styledEmbed()
+        
+        embed.setTitle("Thread Notification")
+        embed.setDescription("A thread has been closed")
+        embed.addFields([
```
Comment (DevScyu) [inline src/events/threadDelete.ts:12]: same as above

**Review by DevScyu — CHANGES_REQUESTED** [review]

File: src/events/threadCreate.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+          name: "Creation Date",
+          value: `<t:${Math.floor(thread.createdTimestamp / 1000)}:R>`,
+          inline: true,
+        },
+      ]);
+
+    thread.join();
+
+    if(logChannel) logChannel.send({ embeds: [embed.toJSON()] });
+    consola.success(`Joined thread ${thread.name}`);
```
Comment (DevScyu) [inline src/events/threadCreate.ts:35]: still need this message?

**Comment by DevChromium** [comment]
moved to https://github.com/Wynntils/Wynntils-Bot/pull/14

**Pull request closed without merging**